lib/ClangImporter/ImportDecl.cpp

```diff
@@ -7904,187 +7904,216 @@ createUnavailableDecl(Identifier name, DeclContext *dc, Type type,
   SmallVector<Decl *, 16> members;
-  llvm::SmallPtrSet<Decl *, 4> knownAlternateMembers;
   for (const clang::Decl *m : objcContainer->decls()) {
     auto nd = dyn_cast<clang::NamedDecl>(m);
-    if (!nd || nd != nd->getCanonicalDecl() ||
-        nd->getDeclContext() != objcContainer) {
-      continue;
+    if (    nd
+        &&  nd == nd->getCanonicalDecl()
+        &&  nd->getDeclContext() == objcContainer) {
+      insertMembersAndAlternates(nd, members);
     }
-
-    forEachDistinctName(nd,
-                        [&](ImportedName name, ImportNameVersion nameVersion) {
-      auto member = importDecl(nd, nameVersion);
-      if (!member)
-        return;
-
-      // If there are alternate declarations for this member, add them.
-      for (auto alternate : getAlternateDecls(member)) {
-        if (alternate->getDeclContext() == member->getDeclContext() &&
-            knownAlternateMembers.insert(alternate).second) {
-          members.push_back(alternate);
-        }
-      }
-
-      // If this declaration shouldn't be visible, don't add it to
-      // the list.
-      if (shouldSuppressDeclImport(nd))
-        return;
-
-      members.push_back(member);
-    });
   }
-
+  
   SwiftDeclConverter converter(*this, CurrentVersion);
+  
+  if (auto clangProto = dyn_cast<clang::ObjCProtocolDecl>(objcContainer))
+    objcContainer = clangProto->getDefinition();
 
-  protos = takeImportedProtocols(D);
-  if (auto clangClass = dyn_cast<clang::ObjCInterfaceDecl>(objcContainer)) {
+  else if (auto clangClass = dyn_cast<clang::ObjCInterfaceDecl>(objcContainer)) {
     auto swiftClass = cast<ClassDecl>(D);
     objcContainer = clangClass = clangClass->getDefinition();
-
+    
     // Imported inherited initializers.
     if (clangClass->getName() != "Protocol") {
       converter.importInheritedConstructors(const_cast<ClassDecl *>(swiftClass),
                                             members);
     }
-
-  } else if (auto clangProto
-               = dyn_cast<clang::ObjCProtocolDecl>(objcContainer)) {
-    objcContainer = clangProto->getDefinition();
   }
-
   // Import mirrored declarations for protocols to which this category
   // or extension conforms.
   // FIXME: This is supposed to be a short-term hack.
   converter.importMirroredProtocolMembers(objcContainer, DC,
-                                          protos, members, SwiftContext);
+                                          takeImportedProtocols(D),
+                                          members,
+                                          SwiftContext);
+  return members;
+}
 
-  // Add the members now, before ~ImportingEntityRAII does work that might
-  // involve them.
-  for (auto member : members) {
-    IDC->addMember(member);
+void ClangImporter::Implementation::loadAllMembersIntoExtension(Decl* D, uint64_t extra) {
+  // We have extension.
+  auto ext = cast<ExtensionDecl>(D);
```

Comment:
Can you use `auto *` here, ala the LLVM style guide:

https://llvm.org/docs/CodingStandards.html#beware-unnecessary-copies-with-auto